packages/flutter_tools/lib/src/device.dart

```diff
@@ -28,6 +28,23 @@ import 'windows/windows_device.dart';
 
 DeviceManager get deviceManager => context.get<DeviceManager>();
 
+/// A broad category the describes the kind of workflow the device supports.
+class Category {
+  const Category._(this.value);
+
+  static const Category web = Category._('web');
+  static const Category fuchsia = Category._('fuchsia');
```

Comment:
As per an email thread, I think we should eliminate this category and the `simulator` below, and for now just have `web`, `mobile`, and `desktop`.